python/dlr/api.py

```diff
@@ -220,7 +220,10 @@ def _get_output_shape(self, index):
             The shape of the index-th output.
         """
         size, dim = self._get_output_size_dim(index)
-        self.output_size_dim.append((size, dim))
+        if index >= len(self.output_size_dim):
```

Comment:
I don't quite understand here. Why index would overflow?

Reply:
In old code, self.output_size_dim would be in correct order only if _get_output_shape(index) gets called with index in consecutive order (0, 1, 2 etc). Furthermore, repeated _get_output_shape() calls will write redundant entries to this list.

Reply:
Discussed offline. No need to have the logic to check and extend the size of the output array. Just resizing to the real number of outputs and then indexing directly should work.